v2/client/client.go

```diff
@@ -42,6 +42,12 @@ func NewAuthClient(voucherURL string) (*Client, error) {
 	return newClient(voucherURL, authClient)
 }
 
+// NewCustomClient creates a new Client set to connect to passed
+// hostname using a passed http client
+func NewCustomClient(voucherURL string, client *http.Client) (*Client, error) {
```

Comment:
Idea: In https://github.com/grafeas/voucher/pull/52 , I was pondering with introducing [functional options](https://golang.cafe/blog/golang-functional-options-pattern.html) as means of wrangling the options for building a client.

I talked myself out of it there, but a third constructor is a sign it's worth considering? I'm a fan of "rule of threes" refactoring.

Reply:
I like the idea of functional options (just read about it), and it seems like a really good approach!